File: salsa/AndroidTheaterStudio/app/src/main/java/com/example/androidtheater5/BatteryBroadcastReceiver.java
package com.example.androidtheater5;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BatteryBroadcastReceiver extends BroadcastReceiver {
    private final static String BATTERY_LEVEL = "level";

    public volatile int currentLevel;

    @Override
    public void onReceive(Context context, Intent intent) {
        currentLevel = intent.getIntExtra(BATTERY_LEVEL, 0);
    }
}

File: opencv-tf/EnergyKotlin/app/src/main/java/com/example/energykotlin/BatteryBroadcastReceiver.kt
package com.example.energykotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent


class BatteryBroadcastReceiver : BroadcastReceiver() {
    @Volatile
    var currentLevel = 0

    override fun onReceive(
        context: Context,
        intent: Intent
    ) {
        currentLevel = intent.getIntExtra(BATTERY_LEVEL, 0)
    }

    companion object {
        private const val BATTERY_LEVEL = "level"
    }
}